fiftyone/utils/hf_hub.py

```diff
@@ -0,0 +1,723 @@
+import json
+import logging
+import os
+from packaging.requirements import Requirement
+from PIL import Image
+import requests
+
+import huggingface_hub as hfh
+import yaml
+
+import fiftyone as fo
+import fiftyone.constants as foc
+from fiftyone.core.config import Config
+import fiftyone.core.dataset as fod
+import fiftyone.core.labels as fol
+from fiftyone.core.sample import Sample
+import fiftyone.core.utils as fou
+
+DATASETS_SERVER_URL = "https://datasets-server.huggingface.co"
+DEFAULT_MEDIA_TYPE = "image"
+DATASET_METADATA_FILENAMES = ("fiftyone.yml", "fiftyone.yaml")
+DATASETS_MAX_BATCH_SIZE = 100
+DEFAULT_IMAGE_FILEPATH_FEATURE = "image"
+FIFTYONE_BUILTIN_FIELDS = ("id", "filepath", "tags", "metadata")
+REPO_TYPE = "dataset"
+SUPPORTED_DTYPES = (
+    "int8",
+    "int16",
+    "int32",
+    "int64",
+    "float16",
+    "float32",
+    "float64",
+    "bool",
+    "string",
+)
+
+logger = logging.getLogger(__name__)
+
+
+def load_from_hub(repo_id, revision=None, **kwargs):
+    """
+    Load a dataset from the Hugging Face Hub into FiftyOne.
+
+    Args:
+        repo_id: the Hugging Face Hub identifier of the dataset
+        revision: the revision of the dataset to load
+        **kwargs: optional keyword arguments
+            split: the split of the dataset to load
+            splits: the splits of the dataset to load
+            subset: the subset of the dataset to load
+            subsets: the subsets of the dataset to load
+            max_samples: the maximum number of samples to load
+            batch_size: the batch size to use when loading samples
+            overwrite (True): whether to overwrite an existing dataset with the same name
+            persistent (False): whether the dataset should be persistent
+            name: the name of the dataset to create
+            config_file: the path to a config file to use to load the dataset,
```

Comment:
Yes! Was thinking about this as well, and for simple cases this would definitely be the better option, but I'd also love to allow config files stored in github repos (for the complicated cases). What do you think about this? I.e. allowing either of the following:

- passing the kwargs directly
- using config from a url